neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -185,6 +196,8 @@ bool IInventory.Verify(Snapshot snapshot)
 
         public virtual bool Verify(Snapshot snapshot, IEnumerable<Transaction> mempool)
         {
+            if (ValidUntilBlock <= snapshot.Height || ValidUntilBlock > snapshot.Height + MaxValidUntilBlockIncrement)
```

Comment:
Why we need to check the second part? `ValidUntilBlock > snapshot.Height + MaxValidUntilBlockIncrement`

Reply:
With this we can receive a `Transaction` and drop it for request again in certains blocks, do you think that is a good check this on `Verify`?

Reply:
I think it is reverified at every block anyway.

Reply:
Maybe it the increment when no parameter is passed.
`tx.ValidUntilBlock = snapshot.Height + Transaction.MaxValidUntilBlockIncrement`

Reply:
https://github.com/neo-project/neo/blob/91df88e85681c89e0dd2abd8f7b38a47d2542bf1/neo/Wallets/Wallet.cs#L56-L58